community/flamingock-auditstore-sql/src/main/java/io/flamingock/community/sql/internal/SqlLockService.java

```diff
@@ -0,0 +1,207 @@
+        }
+        return new LockAcquisition(owner, leaseMillis);
+    }
+
+    @Override
+    public LockAcquisition getLock(LockKey lockKey) {
+        String keyStr = lockKey.toString();
+        try (Connection conn = dataSource.getConnection()) {
+            CommunityLockEntry entry = getLockEntry(conn, keyStr);
+            if (entry != null) {
+                return new LockAcquisition(RunnerId.fromString(entry.getOwner()),
+                        java.sql.Timestamp.valueOf(entry.getExpiresAt()).getTime() - System.currentTimeMillis());
+            }
+        } catch (SQLException e) {
+            // ignore
+        }
+        return null;
+    }
+
+    @Override
+    public void releaseLock(LockKey lockKey, RunnerId owner) {
+        String keyStr = lockKey.toString();
+        try (Connection conn = dataSource.getConnection();
+             PreparedStatement ps = conn.prepareStatement(
+                     "SELECT owner FROM " + lockRepositoryName + " WHERE `key` = ?")) {
+            ps.setString(1, keyStr);
+            try (ResultSet rs = ps.executeQuery()) {
+                if (rs.next()) {
+                    String existingOwner = rs.getString("owner");
+                    if (existingOwner.equals(owner.toString())) {
+                        try (PreparedStatement delete = conn.prepareStatement(
+                                dialectHelper.getDeleteLockSqlString(lockRepositoryName))) {
+                            delete.setString(1, keyStr);
+                            delete.executeUpdate();
+                        }
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            // ignore
+        }
+    }
+
+    private CommunityLockEntry getLockEntry(Connection conn, String key) throws SQLException {
+        try (PreparedStatement ps = conn.prepareStatement(
+                dialectHelper.getSelectLockSqlString(lockRepositoryName))) {
+            ps.setString(1, key);
+            try (ResultSet rs = ps.executeQuery()) {
+                if (rs.next()) {
+                    return new CommunityLockEntry(
+                            rs.getString(1), // key column
+                            LockStatus.valueOf(rs.getString("status")),
+                            rs.getString("owner"),
+                            rs.getTimestamp("expires_at").toLocalDateTime()
+                    );
+                }
+            }
+        }
+        return null;
+    }
+
+    private void upsertLockEntry(Connection conn, String key, String owner, LocalDateTime expiresAt) throws SQLException {
+        String sql = dialectHelper.getInsertOrUpdateLockSqlString(lockRepositoryName);
+
+        if (dialectHelper.getSqlDialect() == SqlDialect.SQLSERVER || dialectHelper.getSqlDialect() == SqlDialect.SYBASE) {
+            // For SQL Server, the SQL already contains transaction management
+            try (Statement stmt = conn.createStatement()) {
```

Comment:
We should try a better option for this, if we can avoid relying on dialect at this point it would be better (it seems not easy).